tasties_app/views.py

```diff
@@ -1,6 +1,54 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.decorators import login_required
+from .forms import CreateUserForm
+from django.contrib import messages
 
 
+@login_required(login_url='login')
 def index(request):
     return render(request, 'tasties_app/index.html',)
-# Create your views here.
+
+
+def base(request):
+    return render(request, 'tasties_app/base.html',)
+
+
+def login_user(request):
+    if request.user.is_authenticated:
+        return redirect('index')
+
+    if request.method == 'POST':
+        username = request.POST.get('username')
+        password = request.POST.get('password')
+        user = authenticate(request, username=username, password=password)
+        if user is not None:
+            login(request, user)
+            return redirect('index')
+        else:
+            messages.info(request, 'Username OR password is incorrect')
```

Comment:
Why do you need these messages? where do you use it?

Reply:
This message will pop up when you insert an invalid username or password on the login page.
It looks like this:

![incorrect](https://user-images.githubusercontent.com/85102059/210235384-610c3ce1-68e3-4d92-a24f-586f0ec79d99.png)

And it comes from these lines in `login.html`:

![Screenshot 2023-01-02 150804](https://user-images.githubusercontent.com/85102059/210235610-97f3d67e-4fb0-4edd-99f2-f38fd8c296cc.png)

Reply:
Thanks